specs/WinRT/WinRTAPIContracts.md

```diff
@@ -0,0 +1,78 @@
+
+
+
+# Windows App SDK API Contracts
+
+## Summary
+Customers trust that Microsoft products are sound.  As Windows App SDK evolves and more developers use Windows App SDK in their own projects Windows App SDK needs a way, at build time and run time, to allow apps to light up API features and to gracefully degrade.  This can be achieved by looking at the deltas between APIs.  
+
+In order to allow apps to depend on API versions Windows App SDK will add contracts to all their WinRT APIs.
+
+
+## Architecture Decisions
+1. Contract boundaries
+   1. One contract per feature/component/sub-system
+   2. Contracts will not span across a transport package/repo.
+2. Contract numbering.
+Contracts will be versioned with one number, starting with 1 and is incremented for any interface change.  This version does not correlate with the Windows App SDK version.
+3. Enforcement
+The build systems will have enforcement gates to make sure that contracts are present, up to date, and there are no conflicts.
+4. Contract Migration.
+Contract Migration will not be a part of this spec.  Windows App SDK will add a spec on contract migration if, and when, there is a situation where it is needed.
+5. Internal contracts
+All WinRT internal APIs will have an contract and the version will always be 1 regardless of any interface changes.
+
+## Adding a new contract
+Developers are encouraged to use their best judgment when deciding where to place a new contract.  If in doubt you can reach out to the Windows App SDK team.
+
+## Example
+
+    namespace Microsoft.Windows.System
+    {
+        [contractversion(2)]
+        apicontract MicrosoftWindowsSystem
+        
+        // version 2
+        [Contract(MicrosoftWindowsSystem(2)]
```

Comment:
i believe contract attribute is commas (as you have on line 50 below but here and on line 43 it's different.
```suggestion
        [Contract(MicrosoftWindowsSystem, 2)]
```

Reply:
From looking at other examples of the contract attribute I don't see any commas. @DrusTheAxe and @BenJKuhn should commas be included, or not in contract versions?

Reply:
Docs say commas

https://docs.microsoft.com/en-us/uwp/midl-3/predefined-attributes#the-contract-attribute
![image](https://user-images.githubusercontent.com/72222/157107855-5eb5aeec-8525-4ff2-9257-b39824b328c9.png)

@BenJKuhn does MIDL also support a Super Secret Secret Squirrel alternative notation of `[contract(Foo(2))]` in addition to the doc'd `[contact(Foo,2)]`?

Reply:
Docs say commas

https://docs.microsoft.com/en-us/uwp/midl-3/predefined-attributes#the-contract-attribute
![image](https://user-images.githubusercontent.com/72222/157108651-0f08b34a-dbe0-4efa-a690-8fb4417f1d24.png)